### PR TITLE
Validates IndexRequests after ingest pipelines (#91777)

### LINKED ISSUE #91777: SetProcessor should prevent user from setting an id longer than 512 bytes
### Description

[IndexRequest](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/action/index/IndexRequest.java#L226) prevents users from setting a document ID more than 512 bytes long because it is not possible to do a get request with an ID that is longer. However users can bypass this by using the SetProcessor like:
```
PUT _ingest/pipeline/test
{
  "processors": [
    {
      "set": {
        "field": "_id",
        "value": "Some_Server_name.thisisareallylongthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamethisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamethisisareallylongsoftwarenamerepeatedthisisareallylongsoftwarenamerepeat"
      }
    }
  ]
}
```
The SetProcessor ought to throw an exception in this case.
Relates to #16034

### PR DESCRIPTION
**Pull request by HiDAl**
This code change checks that the document generated/processed by all pipelines is still valid. If the document will be processed by either 1 or 2 pipelines (namely `pipeline` & `finalPipeline`), the validation would be executed only once per document (`IngestService` receives a "list" of requests, basically a bulk request)

closes #91777

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/ingest/IngestService.java [hunk 1]
```diff
@@ -735,7 +735,9 @@ protected void doRun() {
                         originalThread
                     );
 
-                    i++;
+                    validateIndexRequestPostProcessing(requestSlot, indexRequest, onFailure);
```
Comment (joegallo) [inline server/src/main/java/org/elasticsearch/ingest/IngestService.java:738]: `executePipelines` is (well, can be) asynchronous, you can't call  `validateIndexRequestPostProcessing` here like this.

**Review by joegallo — CHANGES_REQUESTED** [review]
Need to address issues with synchronous/asynchronous code, also I'm not absolutely sure the handler invocation is right (but I'm not sure it's wrong either!).

**Comment by HiDAl** [comment]
PR waiting merge of #92455

**Comment by HiDAl** [comment]
I've updated the PR to the latest version with the fix introduced in #92455. Although this fixes the original issue, it has also surfaced a different kinds of errors:

1. In case you send a `_bulk` request dong upserts **and** the index does not exist yet, the validation won't pass:

```
PUT _ingest/pipeline/lowercase-pipeline
{
  "processors": [
    {
      "lowercase": {
        "field": "my_field"
      }
    }
  ]
}

POST _bulk
{"update":{"_id":"1","_index":"test_index","pipeline":"lowercase-pipeline"}}
{"upsert":{"my_field":"UPPER"},"script":{"source":"ctx._source.my_field = ctx._source.my_field.toLowercase()"}}

{
  "took": 0,
  "ingest_took": 2,
  "errors": true,
  "items": [
    {
      "index": {
        "_index": null,
        "_id": null,
        "status": 400,
        "error": {
          "type": "illegal_argument_exception",
          "reason": "pipeline with id [lowercase-pipeline] does not exist"
        }
      }
    }
  ]
}
```

I'll continue trying to tackle this issue.

**Pull request closed without merging**